### DISCUSSION TITLE
Why SSG generates JSON files?

### DISCUSSION THREAD

**Discussion by Abooddy** [vercel/next.js#35077] [Help]
If an HTML page is pre-generated and contains all the data it should have, why generate these JSON files at `out\_next\data\...` and send them to client?

`{
  "pageProps": {
    "key": "someKey",
    "categoryName": "someCategory",
    "categoryIcon": {
      "prefix": "fas",
      "iconName": "key"
    },
    "categoryIconFlip": null
  },
  "__N_SSG": true
}`

I'm using a local JSON file to dynamically render a single page for over 100 variations using `getStaticProps`. This local JSON file is getting big, and I see it is getting bundled to client so as these framework JSON files for each page.

I searched on this matter but could not find any answer. I would appreciate any help that makes me understand why these files are necessary and if they are not, how I can exclude them from client bundle.

**Comment by chemicalkosek** [comment]
Those come from prefetching. Links in view are being prefeteched to have faster client side routing.
